### PR TITLE
memory leak fix inside track fragment

### LINKED ISSUE #2134: Memory leak inside track fragment
**Actual Behaviour**

Recycler View Instances didn't unregister adapterDataObserver

**Expected Behaviour**

Recycler View should unregister and adapter should be removed

**Would you like to work on the issue?**
yes

### PR DESCRIPTION
**Pull request by simarsingh24**
Fixes #2134

### CODE REVIEW

File: android/app/src/main/java/org/fossasia/openevent/fragments/TracksFragment.java [hunk 1]
```diff
@@ -103,12 +104,13 @@ private void setUpRecyclerView() {
 
         final StickyRecyclerHeadersDecoration headersDecoration = new StickyRecyclerHeadersDecoration(tracksListAdapter);
         tracksRecyclerView.addItemDecoration(headersDecoration);
-        tracksListAdapter.registerAdapterDataObserver(new RecyclerView.AdapterDataObserver() {
+        adapterDataObserver=new RecyclerView.AdapterDataObserver() {
```
Comment (dr0pdb) [inline android/app/src/main/java/org/fossasia/openevent/fragments/TracksFragment.java]: Please add space before and after ```=```

File: android/app/src/main/java/org/fossasia/openevent/fragments/TracksFragment.java [hunk 2]
```diff
@@ -131,6 +133,10 @@ protected int getLayoutResource() {
     public void onDestroyView() {
         super.onDestroyView();
         Utils.unregisterIfUrlValid(this);
+        if(tracksListAdapter!=null){
```
Comment (dr0pdb) [inline android/app/src/main/java/org/fossasia/openevent/fragments/TracksFragment.java]: Add proper spaces here also

**Review by dr0pdb — CHANGES_REQUESTED** [review]

File: android/app/src/main/java/org/fossasia/openevent/fragments/TracksFragment.java [hunk 3]
```diff
@@ -131,6 +133,10 @@ protected int getLayoutResource() {
...
+            if(adapterDataObserver!=null)adapterDataObserver=null;
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/fragments/TracksFragment.java]: No need for a null check here

**Review by iamareebjamal — CHANGES_REQUESTED** [review]

**Review by simarsingh24 — COMMENTED** [review]
@iamareebjamal  if adapterDataObserver is null and is passed to unregisterAdapterDataObserver would that be erroneous? Although we havent set adapterDataObserver to null anywhere but adding an if block certainly assures we are not calling unregisterAdapterDataObserver when adapterDataObserver is null, your views?

**Comment by iamareebjamal** [comment]
@simarsingh24 

```
if(adapterDataObserver!=null)adapterDataObserver=null;
```

Checking object is null just to set it null makes no sense

File: android/app/src/main/java/org/fossasia/openevent/fragments/TracksFragment.java [hunk 4]
```diff
@@ -131,6 +133,9 @@ protected int getLayoutResource() {
     public void onDestroyView() {
         super.onDestroyView();
         Utils.unregisterIfUrlValid(this);
+        tracksListAdapter.unregisterAdapterDataObserver(adapterDataObserver);
+        adapterDataObserver = null;
+        tracksRecyclerView.setAdapter(null);
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/fragments/TracksFragment.java]: Last 2 lines are not needed

**Review by iamareebjamal — CHANGES_REQUESTED** [review]

**Review by iamareebjamal — APPROVED** [review]

**Pull request merged**